haystack/components/tools/tool_invoker.py

```diff
@@ -507,6 +507,10 @@ def run(
                         )
                     )
 
+        # We stream one more chunk that contains a finish_reason if tool_messages were generated
+        if len(tool_messages) > 0 and streaming_callback is not None:
```

Comment:
nit: do we need len here

Reply:
Yes since it's possible that a user sends no actuall tool call messages to the ToolInvoker and when this happens the ToolInvoker just returns an empty list. So if the list is empty it seems kind of odd to also return a streaming chunk with finish reason `tool_call_results` if no tools were actually called.